Refactoring for handling generic classes

VoglSebastian opened the pull request:
Refactoring of CastClassManager + added an interface for a generic class implementation, to change the used implementation of GenericClass

schweikl commented on client/src/main/java/org/evosuite/setup/TestCluster.java:
```diff
@@ -55,6 +52,8 @@
  */
 public class TestCluster {
 
+	// Bei FunctionalInterfaces kann der TestCluster auch MethodenReferenzen oder Lambdas einfügen
```
A comment that should be removed?

schweikl reviewed (commented):
LGTM, some minor things that should be addressed, though (see comments)

schweikl reviewed (commented):
Seems alright now :slightly_smiling_face:

schweikl reviewed (commented):
LGTM

schweikl reviewed (commented):
Is there a particular reason why this Test is now deleted?

VoglSebastian commented:
> Is there a particular reason why this Test is now deleted?

I added this test to verify, that particular methods of TypeUtils do what i assume they do, but i removed the code in which this was assumed. So i deemed this test useless...

schweikl reviewed (commented):
The changes look good to me

schweikl commented:
I guess since all comments are addressed, we are now ready to merge. Unless there is something else you wanted to change?
